src/docfx/lib/path/PackagePath.cs

```diff
@@ -49,7 +49,7 @@ public PackagePath(string remote, string? branch)
         {
             Type = PackageType.Git;
             Url = remote;
-            Branch = branch ?? "master";
+            Branch = AssignBranchWithDefaultBranchFallback(branch);
```

Comment:
Instead of mutating the `branch` value, consider keep the branch value and use a computed property like `IsMasterOrMain`? This way there is no hardcoded `== "main"` check and `DocfxTest.cs` doesn't need to change.